operator/api/v1alpha1/clusterconfig_types.go

```diff
@@ -84,6 +87,57 @@ type ClusterConfig struct {
 	Status ClusterConfigStatus `json:"status,omitempty"`
 }
 
+// Set a condition for the ClusterConfig.
+//
+// If a condition of this type already exists it is updated, otherwise a new
+// condition is added.
+//
+// Arguments
+//
+//	kind:      type of condition
+//	status:    status of the condition
+//	reason:    one-word, CamelCase reason for the transition
+//	message:   details about the transition
+func (c *ClusterConfig) SetCondition(
+	kind string,
+	status metav1.ConditionStatus,
+	reason string,
+	message string,
+) {
+	condition := metav1.Condition{
+		Type:               kind,
+		Status:             status,
+		ObservedGeneration: c.Generation,
```

Comment:
Oh! Is this "standard", i.e. always part of `metav1.Condition`? I'm thinking we may have "over-engineered" https://github.com/scality/zenko-operator/pull/328 (cc @williamlardier) while having this available in each condition allows us to know which one was updated after seeing the CR update... :thinking: Anyway, the current approach should work as well :innocent: but it's good to know!

Reply:
Hey, yes we thought about that for Zenko as well, but we wanted to have it as a standalone value, similar to the deployments, so we can easily check its value regardless of the conditions. But maybe this can be improved, depending on what we want to achieve at the product level.

Reply:
> Oh! Is this "standard", i.e. always part of `metav1.Condition`?

Indeed, and from what I see (but I did not tested myself) the `kubectl wait --for=condition` honor it by default which means that it will wait for the `ObservedGeneration` to match the object `Generation`

Reply:
cc @francoisferrand do you think this could be helpful in a Zenko context?